x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -127,6 +128,50 @@ const degradedDocsRoute = createDatasetQualityServerRoute({
   },
 });
 
+const totalDocsRoute = createDatasetQualityServerRoute({
+  endpoint: 'GET /internal/dataset_quality/data_streams/total_docs',
+  params: t.type({
+    query: t.intersection([
+      rangeRt,
+      typeRt,
+      t.partial({
+        datasetQuery: t.string,
+      }),
+    ]),
+  }),
+  options: {
+    tags: [],
+  },
+  async handler(resources): Promise<{
+    totalDocs: DataStreamDocsStat[];
+  }> {
+    const { context, params } = resources;
+    const coreContext = await context.core;
+
+    const esClient = coreContext.elasticsearch.client.asCurrentUser;
+
+    await datasetQualityPrivileges.throwIfCannotReadDataset(
+      esClient,
+      params.query.type,
+      params.query.datasetQuery
+    );
+
+    const { type, start, end } = params.query;
+
+    const totalDocs = await getAggregatedDatasetPaginatedResults({
+      esClient,
+      start,
+      end,
+      index: `${type}-*-*`,
```

Comment:
Since this es query is the one taking the most time I wanted to keep the types separately.
So the query param is expecting just one type

Reply:
I don't think this would be true for this one since we will get a huge set of documents and try to split them afterwards using `type`, `namespace`, and `dataset`. If we add more types to the request the set will be even bigger. I'd prefer to keep this one per type.
In alignment with what you are saying I just merged the `degradedDocs` using all types at once since there the set of results will be much more controlled.